src/momento/config/transport/grpc_configuration.py

```diff
@@ -12,3 +14,11 @@ def get_deadline(self) -> timedelta:
     @abstractmethod
     def with_deadline(self, deadline: timedelta) -> GrpcConfiguration:
         pass
+
+    @abstractmethod
+    def with_root_certificates_pem(self, root_certificates_pem_path: Path) -> GrpcConfiguration:
+        pass
+
+    @abstractmethod
+    def get_root_certificates_pem(self) -> Optional[bytes]:
+        pass
```

Comment:
I added these to the `GrpcConfiguration` class and implemented in the `StaticGrpcConfiguration` class, which is used across cache, topic, and vector clients. I've only plumbed through using the root certificates in the vector clients though.

As these are buried deep, I feel that most users won't use it; and if one does and needs the functionality in other clients, we can plumb it through there. Otherwise I will either (1) plumb through in other clients now, or (2) specialize a `GrpcConfiguration` subtype that contains the the data and only use that in the vector client.

Reply:
no this is fine